source/common/grpc/common.cc

```diff
@@ -22,7 +22,31 @@
 namespace Envoy {
 namespace Grpc {
 
-const std::string Common::GRPC_CONTENT_TYPE{"application/grpc"};
+bool Common::hasGrpcContentType(const Http::HeaderMap& headers) {
+  const Http::HeaderEntry* content_type = headers.ContentType();
+  if (content_type == nullptr) {
+    return false;
+  }
+  // Fail fast if this is not gRPC.
+  if (!StringUtil::startsWith(content_type->value().c_str(),
+                              Http::Headers::get().ContentTypeValues.Grpc)) {
+    return false;
+  }
+  // Exact match with application/grpc. This and the above case are likely the
+  // two most common encountered.
+  if (Http::Headers::get().ContentTypeValues.Grpc == content_type->value().c_str()) {
```

Comment:
perf nit: if you invert this and compare value() to the constant string you can avoid the string creation.